### PR TITLE
Disable precompiled headers

### PR DESCRIPTION
**Pull request by lava**
We noticed these having an overall negative impact on
build times, since they prevent ccache from caching
the build result.

Since precompiled headers are used for every translation
unit, some minor fixups are needed to fix the build
after disabling them.

### :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

File: libvast/CMakeLists.txt [hunk 1]
```diff
@@ -340,10 +340,10 @@ add_library(vast::native_plugins ALIAS libvast_native_plugins)
 
 # Configure precompiled headers, except for ARM macOS because of this CMake bug:
 # https://gitlab.kitware.com/cmake/cmake/-/issues/21825
-if (NOT (CMAKE_SYSTEM_NAME MATCHES "Darwin" AND CMAKE_SYSTEM_PROCESSOR MATCHES
-                                                "arm"))
-  target_precompile_headers(libvast PUBLIC [["vast/fwd.hpp"]] <caf/fwd.hpp>)
-endif ()
+#if (NOT (CMAKE_SYSTEM_NAME MATCHES "Darwin" AND CMAKE_SYSTEM_PROCESSOR MATCHES
```
Comment (dominiklohmann) [inline libvast/CMakeLists.txt:343]: We could also make this a bit smarter by detecting whether the CXX_COMPILER_LAUNCHER target property is set on libvast.

**Comment by mavam** [comment]
I can confirm that disabling pre-compiled headers makes `ccache` work.

File: libvast/vast/hash/uniquely_represented.hpp [hunk 2]
```diff
@@ -38,10 +39,10 @@ struct is_uniquely_represented<std::tuple<Ts...>>
                        && ((0 + ... + sizeof(Ts)) == sizeof(std::tuple<Ts...>))> {
 };
 
-template <class T, size_t N>
+template <class T, std::size_t N>
```
Comment (mavam) [inline libvast/vast/hash/uniquely_represented.hpp:42]: Why is that needed here?

Comment (lava) [reply libvast/vast/hash/uniquely_represented.hpp:42]: I think something in `fwd.hpp` used to pull `size_t` into the global namespace (e.g. the `<algorithm>` header has that effect), so without it it leads to compile errors from some translation units here. The alternative would be including `fwd.hpp`, but this seems more minimal.

File: libvast/CMakeLists.txt (hunk 1 above)
Comment (lava) [reply libvast/CMakeLists.txt:343]: That's brittle, e.g. on Debian the standard way of using ccache is to install it as a symlink for the compiler binary itself, so it can be used even without cmake support:

```
$ which cc
/usr/lib/ccache/cc
```

If anything, I'd suggest making it an option `VAST_ENABLE_PCH`, but I'm not sure who would be the target audience to enable that option.

File: libvast/vast/hash/uniquely_represented.hpp (hunk 2 above)
Comment (dominiklohmann) [reply libvast/vast/hash/uniquely_represented.hpp:42]: `size_t` is defined in `<cstddef>`. It's guaranteed to be the same type as `std::size_t`.

File: libvast/CMakeLists.txt (hunk 1 above)
Comment (dominiklohmann) [reply libvast/CMakeLists.txt]: Then I'd prefer removing the lines altogether.

File: libvast/vast/hash/uniquely_represented.hpp [hunk 3]
```diff
@@ -9,6 +9,7 @@
 #pragma once
 
 #include <array>
+#include <cstdint>
```
Comment (mavam) [inline libvast/vast/hash/uniquely_represented.hpp:12]:
```suggestion
#include <cstddef>
```

And then change back `std::size_t` to `size_t`, per the other discussion.

**Review by dominiklohmann — APPROVED** [review]
